opnode/eth/heads.go

```diff
@@ -0,0 +1,48 @@
+package eth
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/event"
+)
+
+type HeadSignal struct {
+	Parent BlockID
+	Self   BlockID
+}
+
+// HeadSignalFn is used as callback function to accept head-signals
+type HeadSignalFn func(sig HeadSignal)
+
+// WatchHeadChanges wraps a new-head subscription from NewHeadSource to feed the given Tracker
+func WatchHeadChanges(ctx context.Context, src NewHeadSource, fn HeadSignalFn) (ethereum.Subscription, error) {
```

Comment:
I'm a little less familiar with the internals of the Eth subscription system, but is there a design where the subscription returns the `HeadSignal` events and we don't need a callback?

Reply:
I like channels better than callbacks too, but we're sending it to an `event.Feed` to fan out to multiple channels anyway.
See https://github.com/ethereum-optimism/optimistic-specs/blob/061eb488c2ba02e3d5ec8b70abeda5c9ea6d0645/opnode/node/node.go#L193